In [11]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt 
import seaborn as sns 

from pyfinmod.wacc import wacc, cost_of_debt, cost_of_equity
from pyfinmod.financials import Financials

import tabula

# 1 - What is the WACC for Marriott?

$$
    WACC = (1-\tau) * Cost_{debt} * (\frac{D}{D+E}) + Cost_{equity} * (\frac{E}{D+E})
$$

### 1987 Financial Summary of Marriott

| Total Assets | Long-term Debt | Equity |
| ------------ | -------------- | ------ |
| 5370.5 | 2498.8 | 2871.7 |


In [6]:
E = 2871.7
D = 2498.8
V = E+D 

print(f"Value is {V}, Equity is {E}, and Debt is {D}. \nThe percent of Debt to Value is {D/V}, and the ratio of Debt to Equity is {D/E}")

Value is 5370.5, Equity is 2871.7, and Debt is 2498.8. 
The percent of Debt to Value is 0.4652825621450517, and the ratio of Debt to Equity is 0.8701466030574225


In [8]:
tax_rate = 175.9/398.9
d_to_v = D/V 
d_to_e = D/E 

print(tax_rate)

0.4409626472800201


In [14]:
tabula.read_pdf("S2.1_ACF_Marriott Corporation.pdf", pages=6)[0]

Unnamed: 0     1978     1979     1980     1981  \
0                   Summary of Operations      NaN      NaN      NaN      NaN   
1                                   Sales  1,174.1  1,426.0  1,633.9  1,905.7   
2        Earnings before interest expense      NaN      NaN      NaN      NaN   
3                        and income taxes    107.1    133.5    150.3    173.3   
4                        Interest expense     23.7     27.8     46.8     52.0   
5              Income before income taxes     83.5    105.6    103.5    121.3   
6                            Income taxes     35.4     43.8     40.6     45.2   
7      Income from continuing operationsa     48.1     61.8     62.9     76.1   
8                              Net income     54.3     71.0     72.0     86.1   
9   Funds provided from cont. operationsb    101.2    117.5    125.8    160.8   
10             Capitalization and Returns      NaN      NaN      NaN      NaN   
11                           Total assets  1,000.3  1,080.4  1,214.3  1,454.9   
12                         Total capitalc    826.9    891.9    977.7  1,167.5   
13                         Long-term debt    309.9    365.3    536.6    607.7   
14                Percent to total capita    37.5%    41.0%    54.9%    52.1%   
15                   Shareholders’ equity    418.7    413.5    311.5    421.7   
16               Per Share and Other Data      NaN      NaN      NaN      NaN   
17                    Earnings per share:      NaN      NaN      NaN      NaN   
18                 aContinuing operations      .25      .34      .45      .57   
19                             Net income      .29      .39      .52      .64   
20                         Cash dividends     .026     .034     .042     .051   
21                   Shareholders’ equity     2.28     2.58     2.49     3.22   
22               Market price at year end     2.43     3.48     6.35     7.18   
23       Shares outstanding (in millions)    183.6    160.5    125.3    130.8   
24    Return on avg. shareholders’ equity    13.9%    17.0%    23.8%    23.4%   

       1982     1983     1984     1985     1986     1987  
0       NaN      NaN      NaN      NaN      NaN      NaN  
1   2,458.9  2,950.5  3,524.9  4,241.7  5,266.5  6,522.2  
2       NaN      NaN      NaN      NaN      NaN      NaN  
3     205.5    247.9    297.7    371.3    420.5    489.4  
4      71.8     62.8     61.6     75.6     60.3     90.5  
5     133.7    185.1    236.1    295.7    360.2    398.9  
6      50.2     76.7    100.8    128.3    168.5    175.9  
7      83.5    108.4    135.3    167.4    191.7    223.0  
8      94.3    115.2    139.8    167.4    191.7    223.0  
9     203.6    272.7    322.5    372.3    430.3    472.8  
10      NaN      NaN      NaN      NaN      NaN      NaN  
11  2,062.6  2,501.4  2,904.7  3,663.8  4,579.3  5,370.5  
12  1,634.5  2.007.5  2,330.7  2,861.4  3,561.8  4,247.8  
13    889.3  1,071.6  1,115.3  1,192.3  1,662.8  2,498.8  
14    54.4%    53.4%    47.9%    41.7%    46.7%    58.8%  
15    516.0    628.2    675.6    848.5    991.0    810.8  
16      NaN      NaN      NaN      NaN      NaN      NaN  
17      NaN      NaN      NaN      NaN      NaN      NaN  
18      .61      .78     1.00     1.24     1.40     1.67  
19      .69      .83     1.04     1.24     1.40     1.67  
20     .063     .076     .093     .113     .136      .17  
21     3.89     4.67     5.25     6.48     7.59     6.82  
22    11.70    14.25    14.70    21.58    29.75    30.00  
23    132.8    134.4    128.8    131.0    130.6    118.8  
24    20.0%    20.0%    22.1%    22.1%    20.6%    22.2%

In [21]:
# equity beta of Marriott is 1.11
beta_unl = 1.11 / (1 + d_to_e)
mar_cost_of_debt = 0.0872 + 0.013

print(beta_unl, mar_cost_of_debt)

0.5935363560189927 0.1002


In [28]:
# relever the beta with the new target
tgt_d_to_e = 1.5 

tgt_beta_e = beta_unl * (1+tgt_d_to_e)

In [30]:
print(f"New target equity beta is {tgt_beta_e}")

New target equity beta is 1.4838408900474818


In [36]:
# cost of equity 

cost_of_equity(tgt_beta_e, 0.0872, 0.0743+0.0872)

0.1974493781305279

In [43]:
wacc

<function pyfinmod.wacc.wacc(equity, balance_sheet, income_statement, beta, risk_free_interest_rate, market_return)>

In [44]:
cost_of_equity(tgt_beta_e, 0.0872, 0.0743+0.0872)*0.4 + mar_cost_of_debt*(1-tax_rate)*0.6

0.11258907689773635


### WACC for Marriott
11.26%

# 2 - What is the problem with using WACC of Marriott for evaluating projects in different divisions?